### PR TITLE
refactor(META): upgrade all to Typescript v2 and TSLint 4 

### PR DESCRIPTION
**Pull request by ntilwalli**
TS 2 (v2.03) 
TSLint 4 (4.0.0-dev.0)

This PR updates all repos to the latest stable version of Typescript 2 and updates to the latest
compatible TSLint, it also updates HTTP superagent to v2.3.0 and adds a types property in relevant
package.json files. To be compatible with the latest TSLint the `no-duplicate-keys` needed to be disabled.

- [ ] I added new tests for the issue I fixed/built
- [X] I ran `npm test` for the package I'm modifying
- [X] I used `npm run commit` instead of `git commit`

### CODE REVIEW

**Comment by ntilwalli** [comment]
All tests pass except the SauceLabs step...

**Comment by staltz** [comment]
Thanks @ntilwalli! This is something I wanted to do, go to have a PR for it. I'll review the PR soon.

File: dom/rx-typings.d.ts [hunk 1]
```diff
@@ -1,7 +0,0 @@
-import {EventsFnOptions} from './lib/DOMSource';
-import {Observable} from 'rx';
-export interface DOMSource {
-  select(selector: string): DOMSource;
-  elements(): Observable<Element>;
-  events(eventType: string, options?: EventsFnOptions): Observable<Event>;
-}
```
Comment (staltz) [inline dom/rx-typings.d.ts]: Why remove this? I'm quite sure we still need it, as TS users will `import {DOMSource} from '@cycle/dom/rx-typings'` and also for rxjs, xstream, most.

File: http/most-typings.d.ts [hunk 2]
```diff
@@ -1,6 +0,0 @@
-import {ResponseStream, Response, RequestOptions} from './lib/interfaces';
-import {Stream} from 'most';
-export interface HTTPSource {  
-  filter(predicate: (request: RequestOptions) => boolean): HTTPSource;
-  select(category?: string): Stream<Stream<Response> & ResponseStream>;
-}
```
Comment (staltz) [inline http/most-typings.d.ts]: Also this shouldn't be removed.

File: http/package.json [hunk 3]
```diff
@@ -4,23 +4,25 @@
   "description": "A Cycle.js Driver for making HTTP requests",
   "main": "lib/index.js",
   "typings": "lib/index.d.ts",
+  "types": "lib/index.d.ts",
   "license": "MIT",
   "repository": "https://github.com/cyclejs/cyclejs/tree/master/packages/http",
   "dependencies": {
     "@cycle/xstream-adapter": "3.x.x",
-    "superagent": "2.1.0"
+    "superagent": "2.3.0"
```
Comment (staltz) [inline http/package.json]: Could we separate this version update to a different PR?

File: http/test/common.js [hunk 4]
```diff
@@ -27,7 +27,8 @@ function run(uri) {
         }
 
         var output = Cycle(main, { HTTP: makeHTTPDriver() });
-        output.sources.HTTP.select().mergeAll().subscribe(
+        console.log()
+        const out = output.sources.HTTP.select().mergeAll().subscribe(
```
Comment (staltz) [inline http/test/common.js]: I think this was some artifact left around that you could remove.

File: dom/rx-typings.d.ts (hunk 1 above)
Comment (ntilwalli) [inline dom/rx-typings.d.ts]: Ah, I'll put them back. I was only checking if the build happened and tests passed, and from that perspective these seemed like clutter.

File: http/most-typings.d.ts (hunk 2 above)
Comment (ntilwalli) [inline http/most-typings.d.ts]: Ditto comment above...

File: http/package.json (hunk 3 above)
Comment (ntilwalli) [inline http/package.json]: Sure

**Comment by ntilwalli** [comment]
Fixed

**Comment by staltz** [comment]
Thanks @ntilwalli 
(Finally had time to work on this merge)
